### ISSUE TITLE
Setting mode to fullscreen in settings makes Etterna black screen on macOS.

### ISSUE THREAD

**Issue by bluebandit21** [etternagame/etterna#940]
If display mode is set to "Fullscreen" (Options->Graphics/Sound Options->Display Mode), Etterna will become a fullscreen black screen that doesn't display anything.
As this setting change applies across restarts, the only way to undo this change is to manually edit the Preferences.ini file and set Windowed to 1.

This option should either be disabled for macOS builds, or (preferably) the way we actually fullscreen the game should be fixed for it.

**Comment by bluebandit21** [comment]
I think (might be wrong), that 
```	ADD(ConfOption("Windowed", MovePref<bool>, "Full Screen", "Windowed"));
	g_ConfOptions.back().m_iEffects = OPT_APPLY_GRAPHICS;```
in ScreenOptionsMasterPrefs.cpp might be where this option is added in the game.
If we went with the very low-effort "disable the option on macOS" route, I think you could just through that in a define guard which only triggered for non-macOS systems.

That strikes me more as a temporary fix, though, since we should reasonably support full screen mode on macOS.

**Comment by poco0317** [comment]
In my testing, attempting to remove the option via the c++ this way will cause the screen to crash. When borderless windowed was added, many of the options in this screen were moved to be controlled by lua. So this is actually a lot simpler if you just kind of mess around with ConfDisplayMode() in DisplaySpecs.lua. You can check the running OS there and force the preference setting a certain way instead of following normal rules.

I dont like this

**Comment by bluebandit21** [comment]
Yeah, that feels even more awful on top of an awful temporary patch. 

I guess that lends even more that the best approach would be trying to fix this.

I think that the deprecation below is likely the reason it stopped working.

<img width="682" alt="Screen Shot 2020-12-23 at 7 08 07 PM" src="https://user-images.githubusercontent.com/32799046/103045370-32fcdf80-4552-11eb-9bb9-fb009138995e.png">
